### ISSUE TITLE
injectionError during testing has no useful error message

### ISSUE THREAD

**Issue by juliemr** [angular/angular#15922]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```
This is from current master.

When a DynamicTestModule cannot find a provider, an error is thrown without a useful error message. This happens in unit tests when `TestBed.configureTestingModule` does not contain an import or provider which is later used. It seems that the useful message is getting lost at some point.

Example stack trace:

```
Error
    at injectionError
    at Object.noProviderError 
    at ReflectiveInjector_._throwOrNull 
    at ReflectiveInjector_._getByKeyDefault 
    at ReflectiveInjector_._getByKey 
    at ReflectiveInjector_.get 
    at DynamicTestModuleInjector.NgModuleInjector.get 
    at resolveDep
```

cc @mhevery

**Comment by juliemr** [comment]
After a bit of digging, it looks like `error.message` is initially good, but gets lost somewhere between jasmine and zones.

**Comment by riturajratan** [comment]
Have you found any solution

**Comment by juliemr** [comment]
ping @mhevery @vikerman

**Comment by juliemr** [comment]
Related: It would be useful to have a chain of errors showing where the missing service is being requested.

**Comment by vikerman** [comment]
This seems to have been fixed by ce1d7c4
